### PR TITLE
chore(instance, sidebar): Add navigation items for databases and queries tab to the sidebar COMPASS-5410

### PR DESCRIPTION
**Pull request by gribnoysup**
## Description

Adds new navigation items that open "Databases" and "My Queries" tabs in the instance workspace when clicked. The change is hidden behind `COMPASS_SHOW_YOUR_QUERIES_TAB` env var until the whole feature is ready. This is how it looks:

Kapture 2022-01-19 at 10 28 35

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [x] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass-sidebar/src/components/sidebar/navigation-items.jsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+              icon={item.icon}
+              onItemClick={() => onItemClick(item.tabName)}
+              isExpanded={isExpanded}
+            />
+          );
+        })}
+      </ul>
+    </nav>
+  );
+};
```
Comment (gribnoysup) [inline packages/compass-sidebar/src/components/sidebar/navigation-items.jsx:93]: In theory this should rather be part of the same navigation keyboard flow as selecting the database / collection, but this would require a much more involved refactoring and this sidenav is being reworked in the future anyway, tabbable buttons seem like an passable compromise for now

File: packages/compass-sidebar/src/stores/store.js [hunk 2]
```diff
@@ -125,7 +125,7 @@ store.onActivated = (appRegistry) => {
     store.dispatch(changeActiveNamespace(dbName));
   });
 
-  appRegistry.on('select-instance', () => {
+  appRegistry.on('sidebar-instance-workspace-open-tab', () => {
```
Comment (mcasimir) [inline packages/compass-sidebar/src/stores/store.js:128]: super non-blocking .. what does this name stand for? Is it more something like `workspace-open-tab-from-sidebar`? Why it was `select-instance` before?

File: packages/compass-home/src/components/home.tsx [hunk 3]
```diff
@@ -219,7 +219,7 @@ function Home({ appName }: { appName: string }): React.ReactElement | null {
     appRegistry.on('data-service-disconnected', onDataServiceDisconnected);
     appRegistry.on('select-database', onSelectDatabase);
     appRegistry.on('select-namespace', onSelectNamespace);
-    appRegistry.on('select-instance', onSelectInstance);
+    appRegistry.on('sidebar-instance-workspace-open-tab', onSelectInstance);
```
Comment (mcasimir) [inline packages/compass-home/src/components/home.tsx:222]: should we also rename the handler to match the event name?

File: packages/compass-components/src/hooks/use-focus-ring.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+});
+
+export const focusRingVisible = css({
+  '&::after': {
+    boxShadow: `0 0 0 3px ${uiColors.focus}`,
+    transitionTimingFunction: 'ease-out',
+  },
+});
+
+export function useFocusRing<T extends HTMLElement>(): React.HTMLProps<T> {
```
Comment (mcasimir) [inline packages/compass-components/src/hooks/use-focus-ring.ts:31]: Nice!

File: packages/compass-sidebar/src/stores/store.js (hunk 2 above)
Comment (gribnoysup) [reply packages/compass-sidebar/src/stores/store.js:128]: It emits from the sidebar, so I put the sidebar in the event name as the first thing similar to how I done it [in the databases navigation section](https://github.com/mongodb-js/compass/blob/199d42553a2b3f8af37bed621476952f86847a5f/packages/compass-sidebar/src/components/sidebar-databases-navigation/index.js#L58-L66). Should I change it to something else there too while I'm at it? I can also keep it as `select-instance` if you feel like it's better

File: packages/compass-home/src/components/home.tsx (hunk 3 above)
Comment (gribnoysup) [reply packages/compass-home/src/components/home.tsx:222]: Yeah, totally, somehow missed it, thanks for highlighting this

File: packages/compass-sidebar/src/stores/store.js (hunk 2 above)
Comment (mcasimir) [reply packages/compass-sidebar/src/stores/store.js:128]: Just to be clear I think we can go on with `sidebar-instance-workspace-open-tab` too.

`select-instance` maybe at this point is a bit too generic, at least I would not understand what it does now. What does "instance" stands for here?

A convention where we add the source in the event could be a great idea, but at the moment is not very clear, it just seems like any other word since everything is separated by `-` and that is not something happening for all the events. Would be different if the first separator is different somehow: `sidebar/instance-workspace-open-tab`.

How would you describe this action in plain English?

**Review by mcasimir — APPROVED** [review]

File: packages/compass-sidebar/src/stores/store.js (hunk 2 above)
Comment (gribnoysup) [reply packages/compass-sidebar/src/stores/store.js:128]: So `instance` is how Compass refers to the instance you are connected to (see `instance-model` or `Instance.Workspace`), I agree that it's maybe not the best name, but I'm not inventing new terms here, just trying to keep it consistent with other parts of the app, I added workspace to the name so it's a bit clearer that we are not actually selecting a different instance.

FWIW you would usually use `:` as a delimiter and compass even does it sometimes (but not everywhere and I honestly struggle to find a pattern, seems to be mostly done in telemetry? 🤷 ) and I'm just not sure what to use as there is no conventions anywhere and mixing two in one place (other events emitted from sidebar don't have any delimiter) felt weird.

Probably something like `open instance workspace with tab n selected (optionally)`, so the event might be `open-instance-workspace` and tabName as an argument? Makes sense?

Comment (gribnoysup) [reply packages/compass-sidebar/src/stores/store.js]: Renamed it to `open-instance-workspace` and will merge now, but please tell me if you want me to change the event name to something else
